### ISSUE TITLE
Plans for docker-machine support?

### ISSUE THREAD

**Issue by fastnsilver** [rhuss/docker-maven-plugin#264]
Docker is deprecating `boot2docker`. See message when you `boot2docker up` with v1.8.0.

What plans do you have for supporting `docker-machine`?  See https://www.docker.com/docker-machine and https://docs.docker.com/machine/get-started/.

**Comment by jgangemi** [comment]
i'm not sure the two are related here. you would use `docker-machine` to achieve the same thing that the boot2docker setup was doing for you, set up a `VirtualBox` instance that will run `docker-engine` that you can then deploy containers to.

you would then configure the `docker-maven-plugin` to point to that `VirtualBox` host, similar to how `boot2docker` works now, and run the plugin accordingly.

could you describe a use-case you see the plugin fulfilling for `docker-machine`?

**Comment by fastnsilver** [comment]
Thanks for the quick reply. It's just the out of the box experience I'm bemoaning here.  If someone (like me ;) ) walks up to use this plugin, what's the minimum config necessary to achieve the same results with a `Dockerfile` and `docker build...` via `docker-machine`?  

My project does not employ a `Dockerfile` rather the config as per this plugin (though I know the former is an option).  I did NOT know I could point the `docker-maven-plugin` at my VBox image. (How and where would I do this? properties?)  

I'll attempt to illustrate what lead me to write-up this issue...

When I had uninstalled `boot2docker` and installed `docker-machine` the `docker-maven-plugin` `docker:build` goal complained about a missing `boot2docker`. With `boot2docker` installed this Maven plugin works without any further need of config (it knows where to source the VM hosting Docker).

**Comment by rhuss** [comment]
By default the plugin uses the environment variable `DOCKER_HOST` which should be set externally like with `eval $(boot2docker shellinit)` or `$(docker-machine env default)` (when `default` it the docker machine's name). The plugin has been tested with a docker-machine setup and it works well.

So, the plugin needs the same setup like the docker CLI (regardless whether you are using docker-machine or boot2docker). BTW, there is no direct reference at all to boot2docker  from within the plugin, so I wonder what is complaining about boot2docker in your case. Could you lease post some logs ?

I agree with you that a 'self-contained' is a good goal, but even when you are using docker-machine you still need to install docker-machine itself and setup these environment variables for you with the commands given above.

And I also agree with @jgangemi that calling docker-machine from within the plugin is out of scope ;-)

**Comment by jgangemi** [comment]
yeah - it only knows the host b/c `boo2docker` spawns a new terminal window w/ the proper environment variables already set which the plugin can then pick up if you haven't explicitly set it in the configuration.

for all intents and purposes, `docker-machine` == `boot2docker`, it just seems there are a few manual steps you have to perform when setting up via `docker-machine` that `boot2docker` did for you automatically.

i run a vmware instance on my mac that is my docker host. `boot2docker` and `docker-machine` were ever involved in its creation (although i think `docker-machine` can create vmware instances now).

in my `.bashrc`, i have this entry:
```
export DOCKER_HOST=tcp://172.16.116.139:2375
```
where `172.16.116.139` is the ip address of my linux vmware instance and port `2375` is the docker port (yes, i'm being bad and not running ssl but i don't really care on my laptop).

any terminal window can automatically talk to the docker daemon b/c of that env variable. the `docker-maven-plugin` picks up that value as well.

this plugin also knows nothing about `boot2docker`, so the error you got was probably a slight red herring. could you post the error here? i'm curious what it said.

**Comment by fastnsilver** [comment]
@rhuss Indeed! I had forgotten that I had manually set `DOCKER_HOST`, `DOCKER_CERT_PATH`, and `DOCKER_TLS_VERIFY` in my `.bash_profile` on my Mac.  Now, I have updated that file to include calls to `eval $(boot2docker shellinit)` and `eval $(docker-machine env dev)` within separate functions so I can switch between using `boot2docker` and `docker-machine` managed instances at will.

e.g.,

```
    use-boot2docker () {
        eval $(boot2docker shellinit)
    }
    
    use-docker-machine () {
        eval $(docker-machine env dev)
    }
```

I apologize for what amounts to user error. The title of this issue is a bit of a misnomer.  I was hamstrung by not remembering to update my environment configuration to support Docker.  But now I'm good.  Thanks!
